### PR TITLE
Add text conventions document

### PR DESCRIPTION
**Pull request by maciaszczykm**
Regarding https://github.com/kubernetes/dashboard/issues/1223#issuecomment-246650072.

### CODE REVIEW

**Comment by maciaszczykm** [comment]
Feel free to post your comments to make it better.

**Comment by maciaszczykm** [comment]
Will merge when we will solve CLA issue.

File: docs/devel/text-conventions.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Text conventions
+
+The document describes the proper way to introduce new text to the Kubernetes Dashboard.
+
+## Overview
+Kubernetes Dashboard is a web-based user interface for Kubernetes clusters. It consists of multiple views like overview pages, list of resources and resource details pages. Much textual information is required within these elements. Let's consider sample dialog:
+
+![Dashboard dialog](../dashboard-ui.png)
+
+As you noticed there is a lot of different kinds of text messages in each view:
```
Comment (digitalfishpond) [inline docs/devel/text-conventions.md]: 'there are a lot...'

File: docs/devel/text-conventions.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- titles
+- headers
+- labels
+- values
+- tooltips
+- action button
+- menu and navigation entries
+- all kinds of messages including warnings, errors and help information
+
+For each one of them developer has to take following decisions:
```
Comment (digitalfishpond) [inline docs/devel/text-conventions.md]: ...them a developer has to make the following...

File: docs/devel/text-conventions.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+- Punctuation - Using a period or not?
+- Capitalization - Using capitalized words? Capitalize Kubernetes specific names?
+
+In addition all text messages within Dashboard should be capitalized and it should be taken into consideration. For more details check our [localization guidelines](localization.md).
+
+## General terms
+
+### Grammar
+
+Gerund form should be used everywhere. Exceptions are all kinds of messages including warnings, errors and help information.
```
Comment (digitalfishpond) [inline docs/devel/text-conventions.md]: The gerund form

File: docs/devel/text-conventions.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+### Punctuation
+
+Depending on the type of a text resource, periods should be used at the end of sentences or not. Periods should be used within:
```
Comment (IanLewis) [inline docs/devel/text-conventions.md]: "Certain types of text resources should have punctuation. Periods should be used within:"

Comment (digitalfishpond) [inline docs/devel/text-conventions.md]: within --> for (within seems strange as it means 'inside of' which is technically ok of course, but a period comes at the end of a sentence, not inside)

File: docs/devel/text-conventions.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+- all kinds of messages including warnings, errors and help information
+
+Periods should be avoided within:
```
Comment (digitalfishpond) [inline docs/devel/text-conventions.md]: within --> for

File: docs/devel/text-conventions.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+- headers
+- titles
+- labels
+- values
+- tooltips
+- menu and navigation entries
+
+### Capitalization
+
+All kinds of the messages should have their first word capitalized. Exceptions are:
```
Comment (IanLewis) [inline docs/devel/text-conventions.md]: "In general, all kinds of messages should have their first word capitalized."

Comment (digitalfishpond) [inline docs/devel/text-conventions.md]: kinds of messages

File: docs/devel/text-conventions.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+- names which appear in the middle of a message
+- values and table contents
+
+Moreover, Kubernetes specific names should be capitalized everywhere. It includes:
```
Comment (IanLewis) [inline docs/devel/text-conventions.md]: "This includes:"

Comment (digitalfishpond) [inline docs/devel/text-conventions.md]: Kubernetes-specific

File: docs/devel/text-conventions.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+The document describes the proper way to introduce new text to the Kubernetes Dashboard.
+
+## Overview
+Kubernetes Dashboard is a web-based user interface for Kubernetes clusters. It consists of multiple views like overview pages, list of resources and resource details pages. Much textual information is required within these elements. Let's consider sample dialog:
+
+![Dashboard dialog](../dashboard-ui.png)
+
+As you noticed there is a lot of different kinds of text messages in each view:
+
+- titles
```
Comment (IanLewis) [inline docs/devel/text-conventions.md]: nit: I would capitalize the items in lists.

File: docs/devel/text-conventions.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+- menu and navigation entries
+- all kinds of messages including warnings, errors and help information
+
+For each one of them developer has to take following decisions:
+
+- Grammar - Using a verb in infinitive or gerund form?
+- Punctuation - Using a period or not?
+- Capitalization - Using capitalized words? Capitalize Kubernetes specific names?
+
+In addition all text messages within Dashboard should be capitalized and it should be taken into consideration. For more details check our [localization guidelines](localization.md).
```
Comment (IanLewis) [inline docs/devel/text-conventions.md]: Not sure what you mean here. It looks like you addressed capitalization above in the list above?

Do you mean localized? If so, "In addition, with few exceptions, all text messages within the Dashboard should be localized."

**Comment by maciaszczykm** [comment]
Thank you guys for your input :) I'll reply soon.

**Comment by maciaszczykm** [comment]
CLA test.

**Comment by maciaszczykm** [comment]
CLA test.

File: docs/devel/text-conventions.md (hunk 1 above)
Comment (maciaszczykm) [inline docs/devel/text-conventions.md]: Done.

File: docs/devel/text-conventions.md (hunk 7 above)
Comment (maciaszczykm) [inline docs/devel/text-conventions.md]: Done.

Comment (maciaszczykm) [inline docs/devel/text-conventions.md]: Done.

File: docs/devel/text-conventions.md (hunk 6 above)
Comment (maciaszczykm) [inline docs/devel/text-conventions.md]: Done.

Comment (maciaszczykm) [inline docs/devel/text-conventions.md]: Done.

File: docs/devel/text-conventions.md (hunk 5 above)
Comment (maciaszczykm) [inline docs/devel/text-conventions.md]: Done.

File: docs/devel/text-conventions.md (hunk 4 above)
Comment (maciaszczykm) [inline docs/devel/text-conventions.md]: Done.

Comment (maciaszczykm) [inline docs/devel/text-conventions.md]: Done.

File: docs/devel/text-conventions.md (hunk 3 above)
Comment (maciaszczykm) [inline docs/devel/text-conventions.md]: Done.

File: docs/devel/text-conventions.md (hunk 2 above)
Comment (maciaszczykm) [inline docs/devel/text-conventions.md]: Done.

File: docs/devel/text-conventions.md (hunk 8 above)
Comment (maciaszczykm) [inline docs/devel/text-conventions.md]: This seems to be a matter of style, but I've switched it. Actually I like to capitalize items in lists only if I'm using full sentences.

File: docs/devel/text-conventions.md (hunk 9 above)
Comment (maciaszczykm) [inline docs/devel/text-conventions.md]: Yes, of course. Good catch :)

**Comment by maciaszczykm** [comment]
PTAL

**Comment by maciaszczykm** [comment]
Thank you for review, merged.

**Pull request merged**